drv.c

```diff
@@ -140,6 +140,14 @@ struct driver *drv_create(int fd)
 	if (!drv->combos)
 		goto free_mappings;
 
+	if (drv->backend->init) {
+		ret = drv->backend->init(drv);
+		if (ret) {
+			drv_array_destroy(drv->combos);
+			goto free_mappings;
+		}
+	}
+
```

Comment:
The reason why seperated 'create()' and 'init()'  before is that: inside "i915_add_combinations()" function, it'll need this check (which will check igpu/dgpu co-exists):
		// in dual gpu case, only alloc x-tiling for dgpu for render
		if ((drv->gpu_grp_type & GPU_TYPE_DUAL_IGPU_DGPU) && (i915->graphics_version >= 125))
			return 0;
If this condition can be replaced, then it's not needed to seperate 'create()'/'init()'

Reply:
Code updated.